File: fastman-challenge/fastman-java/src/test/java/com/josdem/challenge/FastManFinderTest.java
package com.josdem.challenge;

/*
Given A string with no spaces
And a source dictionary as words in an array
When I call to find method
Then I validate expected words are found
 */

import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertEquals;

public class FastManFinderTest {

    private final FastManFinder fastmanFinder = new FastManFinder();

    @Test
    @DisplayName("It finds fast and man words")
    void shouldFindTwoWords() {
        assertEquals("fast man", fastmanFinder.find("fastman"));
    }

}

File: fastman-challenge/fastman-kotlin/src/test/kotlin/com/josdem/challenge/FastManFinderTest.kt
package com.josdem.challenge

/*
Given A string with no spaces
And a source dictionary as words in an array
When I call to find method
Then I validate expected words are found
 */

import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Test

class FastManFinderTest {
    private val fastManFinder: FastManFinder = FastManFinder()

    @Test
    fun `it finds fast and man words`() {
        assertEquals("fast man", fastManFinder.find("fastman"));
    }
}